src/gatsby/plugins/gatsby-remark-terms/index.js

```diff
@@ -25,22 +25,28 @@ const CUSTOM_LINK_START = new RegExp("^<([a-zA-Z]+Link|a) ");
 const CUSTOM_LINK_END = new RegExp("^</([a-zA-Z]+Link|a)>");
 
 function replace(node) {
+  if (node.type == "root") return;
+
   // If this is an empty node there's nothing to consider.
-  if (!node.children) return;
+  if (!node.children) return "skip";
 
   // Do not replace abbreviations in headings because that appears to break the heading anchors.
-  if (node.type == "heading") return;
+  if (node.type == "heading") return "skip";
 
   // Do not replace abbreviations in links because that's two interactive
   // nested elements nested in each other, and we decided we don't want to
   // style that either.
   //
   // This currently doesn't handle nesting of e.g.
   // <a><strong><abbr>... but we don't have that in docs.
-  if (node.type == "link") return;
+  if (node.type == "link") return "skip";
 
   let insideCustomLink = false;
 
+  const alreadyExplained = {};
+
+  const newChildren = [];
```

Comment:
When I made the initial change, the build never terminated. I had to run Instruments.app against gatsby to figure that most of the time was spent in array splicing. Fixing this gross misuse of datastructures made the build terminate again.

It seems that the second-heaviest stacktrace is about regex replacements. Will create a followup PR.

Reply:
sike, the thing i wanted to do doesn't actually work. I think we would want to spend some time thinking about how to make gatsby-remark-variables faster, which I think is dominating our build times. cc @dcramer